### PR TITLE
refactor: remove pythnet feature flag

### PR DESCRIPTION
**Pull request by cctdaniel**
solana specific code is duplicated in https://github.com/pyth-network/pyth-client/tree/solana so we have a fallback in case we need to make any fix

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,10 +155,10 @@ Finally, in docker extension inside VS Code click right and choose "Attach VS Co
 
 Oracle program upgrades are managed by the Pythian Council multisig. The steps to deploy a new version are:
 
-1. Create a release branch from `main`. This should include binaries for both the Solana mainnet and Pythnet oracle programs (`pyth_oracle_solana.so` and `pyth_oracle_pythnet.so`).
+1. Create a release branch from `main`. This should include binary for both the Pythnet oracle programs (`pyth_oracle_pythnet.so`).
```
Comment (ali-bahjati) [inline README.md:158]: for both the Pythnet oracle programs?

File: program/rust/src/tests/pyth_simulator.rs [hunk 2]
```diff
@@ -78,11 +78,7 @@ lazy_static::lazy_static! {
 
     // Detect features and pass them onto cargo-build-bpf.
     // IMPORTANT: All features of this crate must have gates added to this vector.
-    let features: Vec<&str> = vec![
-    #[cfg(feature = "pythnet")]
-    "pythnet",
-
-    ];
+    let features: Vec<&str> = vec!["pythnet"];
```
Comment (ali-bahjati) [inline program/rust/src/tests/pyth_simulator.rs:81]: we won't need it anymore right?

File: README.md (hunk 1 above)
Comment (cctdaniel) [reply README.md:158]: good catch

File: README.md [hunk 3]
```diff
@@ -155,10 +155,10 @@ Finally, in docker extension inside VS Code click right and choose "Attach VS Co
 
 Oracle program upgrades are managed by the Pythian Council multisig. The steps to deploy a new version are:
 
-1. Create a release branch from `main`. This should include binaries for both the Solana mainnet and Pythnet oracle programs (`pyth_oracle_solana.so` and `pyth_oracle_pythnet.so`).
+1. Create a release branch from `main`. This should include binary for the Pythnet oracle program (`pyth_oracle_pythnet.so`).
```
Comment (guibescos) [inline README.md:158]: nit : this should include **the** binary

File: program/rust/src/tests/pyth_simulator.rs (hunk 2 above)
Comment (guibescos) [reply program/rust/src/tests/pyth_simulator.rs:81]: Yes you can get rid of the whole `lazy_static`

**Review by guibescos — COMMENTED** [review]
Another file I'd try to cleanup is `build.rs`

File: program/rust/src/processor/upd_price.rs [hunk 4]
```diff
@@ -197,7 +183,6 @@ pub fn upd_price(
     let mut price_data = load_checked::<PriceAccount>(price_account, cmd_args.header.version)?;
 
     // Feature-gated accumulator-specific code, used only on pythnet/pythtest
-    #[cfg(feature = "pythnet")]
```
Comment (guibescos) [inline program/rust/src/processor/upd_price.rs:200]: I think a good refactor you can already to here is move the block :
`        if aggregate_updated {
            price_data.message_sent_ = 0;
            price_data.update_price_cumulative()?;
        }`
closer to the `c_upd_twap` code

File: program/c/src/oracle/oracle.h [hunk 5]
```diff
@@ -27,12 +27,9 @@ extern "C" {
 #define PC_NUM_COMP_PYTHNET     128
 
 // PC_NUM_COMP - number of price components in use
-#ifdef PC_PYTHNET
 // Not whole PC_NUM_COMP_PYTHNET because of stack issues appearing in upd_aggregate()
 #define PC_NUM_COMP 64
-#else
-#define PC_NUM_COMP PC_NUM_COMP_SOLANA
```
Comment (guibescos) [inline program/c/src/oracle/oracle.h:34]: maybe delete `PC_NUM_COMP_SOLANA`

File: program/rust/src/tests/pyth_simulator.rs [hunk 6]
```diff
@@ -139,7 +102,19 @@ struct ProductMetadata {
 impl PythSimulator {
     /// Deploys the oracle program as upgradable
     pub async fn new() -> PythSimulator {
-        let mut bpf_data = read_file(&*ORACLE_PROGRAM_BINARY_PATH);
+        let mut cmd = std::process::Command::new("cargo");
```
Comment (guibescos) [inline program/rust/src/tests/pyth_simulator.rs:105]: I think the right way of solving this is replacing `cargo test` by `cargo-test-bpf` in `build-bpf.sh`.
That will internally call `cargo-build-bpf` before starting the tests.

**Review by guibescos — DISMISSED** [review]

File: program/c/src/oracle/oracle.h (hunk 5 above)
Comment (cctdaniel) [reply program/c/src/oracle/oracle.h:34]: I think this is still used in `PC_EXPECTED_PRICE_T_SIZE_PYTHNET`

File: scripts/build-bpf.sh [hunk 7]
```diff
@@ -20,7 +20,7 @@ set -x
 cd "${PYTH_DIR}"
 
 # Re-run tests affected by features
-cargo test --locked
+cargo-test-bpf
```
Comment (cctdaniel) [inline scripts/build-bpf.sh:23]: `cargo-test-bpf` doesn't work with `-- --locked`

File: program/c/src/oracle/oracle.h (hunk 5 above)
Comment (ali-bahjati) [reply program/c/src/oracle/oracle.h:34]: So perhaps remove it from there as well? :?

Comment (guibescos) [reply program/c/src/oracle/oracle.h:34]: Yep

**Review by guibescos — DISMISSED** [review]

File: program/c/src/oracle/oracle.h [hunk 8]
```diff
@@ -208,13 +207,45 @@ typedef struct pc_price
...
+Adding up all these sizes gives us a total of 240 bytes for the fixed part of the pc_price_t struct.
+The size of the comp_ array is variable and depends on PC_NUM_COMP, hence it is calculated separately and added to the base size of 240 bytes.
+*/
+#define PC_EXPECTED_PRICE_T_SIZE_PYTHNET (240 \
 					+ PC_NUM_COMP * sizeof(pc_price_comp_t) \
 					)
 
 static_assert( sizeof( pc_price_t ) == PC_EXPECTED_PRICE_T_SIZE_PYTHNET, "" );
-#undef PC_EXPECTED_PRICE_SIZE_PYTHNET
+#undef PC_EXPECTED_PRICE_T_SIZE_PYTHNET
```
Comment (cctdaniel) [inline program/c/src/oracle/oracle.h:248]: this seems like a typo before

**Review by ali-bahjati — APPROVED** [review]

**Review by guibescos — APPROVED** [review]
send it

**Pull request merged**